auth/auth.go

```diff
@@ -25,6 +26,20 @@ func Authenticate(req *http.Request) (models.User, error) {
 	return user, err
 }
 
+// Authenticate admin returns an error if the given request basic auth is unable to authenticate admin user, nil otherwise
+func AuthenticateAdmin(req *http.Request) (models.User, error) {
+	username, password, ok := req.BasicAuth()
+	if !ok {
+		return models.User{}, errors.New("Unable to get request basic auth")
+	}
+
+	var err error
+	if username != viper.Get("admin.username") && password != viper.Get("admin.passowrd") {
```

Comment:
Please use viper.GetString instead of Get, otherwise you'll get an interface (and you need to cast it to really check the value).